### PR TITLE
DITTO app

### PR DESCRIPTION
**Pull request by tkmamidi**
The DITTO application provides a user-friendly interface for analyzing and interpreting genetic variants in build GRCh38 format (chr, pos, ref, alt). Leveraging the power of openCravat's API, DITTO seamlessly retrieves annotations for the input variant, enriching the analysis with valuable biological context.

At the core of DITTO lies the DITTO model, a robust explainable machine learning model trained to predict the deleteriousness score of a given variant. This score quantifies the likelihood of a variant being likely pathogenic.

To further enhance the interpretability of DITTO's predictions, the application employs SHAP (SHapley Additive exPlanations), a technique that provides detailed explanations for each prediction. These explanations reveal the specific features that contribute to the model's decision-making process, offering valuable insights into the variant's potential impact.

Please review -
- [ ] Follow ReadMe and verify install instructions
- [ ] Spin up the app locally and verify if everything is working.

### CODE REVIEW

**Comment by ManavalanG** [comment]
Streamlit cloud is using `pip`, not conda, to install the dependencies, as `requirements.txt` is used to supply dependencies. I would recommend matching env creation in readme to match the env used in streamlit cloud. If there are time concerns, at the least create an issue and then add a warning note in readme mentioning this discrepancy.

image

**Comment by ManavalanG** [comment]
As shown in the screenshot in [another comment here](https://github.com/uab-cgds-worthey/DITTO-UI/pull/11#issuecomment-1832050728), streamlit cloud spends significant amount of time cloning the repo, which is undesirable as it makes it harder to debug, develop and test when one has to spin up the webapp. As the files that were causing huge `.git` size are not needed anymore, it may be a good idea to remove them from git history if they are not needed again in the future. 

Note that this is a recommendation and not a requirement. Please consult Brandon if you choose to scrub them from git history.

**Comment by ManavalanG** [comment]
Webapp's name in the browser tab shows up wrongly as `DITTO4NF`.
image

**Comment by ManavalanG** [comment]
Including a short description in webapp for the following would be really helpful for the users:

* what ditto is
* How to interpret ditto scores
* How opencravat gets used

**Comment by ManavalanG** [comment]
Ditto's description in the webapp states `A tool for exploring likely pathogenic variants...`. Is it really limited to just the likely-pathogenic variants? Please clean up as needed.

**Comment by ManavalanG** [comment]
The webapp says "Please feel free to contact us with any issues, comments, or questions". In addition, I would recommend linking it to ditto-ui's github as it provides a natural mechanism to create and follow issues.

**Comment by ManavalanG** [comment]
Nitpicking: Section `Local Installation and Setup` is a bit confusing. It discusses fetching the source code but it lists conda as a requirement. I would recommend cleaning up the section (content or/and title).

**Review by ManavalanG — CHANGES_REQUESTED** [review]

File: ReadMe.md [hunk 1]
```diff
@@ -7,13 +7,13 @@ _!!! For research purposes only !!!_
 
 ## Description
 
-A web app where anyone can lookup variants and understand the mechanism/details of these variants such as domain,
+A web app where anyone can lookup variants and understand the mechanism/details of these variants such as
 function, DITTO deleterious score and Clinvar reported significance.
```
[the remark this answers is not in the document]
Comment (tkmamidi) [reply ReadMe.md]: done.

File: ReadMe.md [hunk 2]
```diff
@@ -7,13 +7,13 @@ _!!! For research purposes only !!!_
...
 
 ## Data
 
-DITTO-UI comprises precomputed DITTO scores for missense variants using dbNSFP data (downloaded on 03/19/2022) for
-training DITTO and making predictions. Domain information is extracted from Uniprot via an API query on page render.
+DITTO-UI comprises precomputed DITTO scores for any variant using annotations from openCravat
```
[the remark this answers is not in the document]
Comment (tkmamidi) [reply ReadMe.md]: fixed it.

File: ReadMe.md [hunk 3]
```diff
@@ -1,71 +1,74 @@
...
-git clone https://github.com/uab-cgds-worthey/DITTO-UI.git
+    git clone https://github.com/uab-cgds-worthey/DITTO-UI.git
 ```
 
-#### Environement Creation
+### Requirements
+
+*OS:*
+
+Currently works only in Mac OS. Docker versions may need to be explored later to make it useable in Mac (and
```
Comment (ManavalanG) [inline ReadMe.md:40]: >useable in Mac

Did you mean to say linux? Btw, I'm surprised it wouldn't work in both Mac and Linux.

**Review by ManavalanG — CHANGES_REQUESTED** [review]

File: ReadMe.md [hunk 4]
```diff
@@ -1,71 +1,74 @@
...
 
 Change in to root directory and run the commands below:
 
 ```sh
-# create environment with mamba
-mamba env create -n ditto-env -f requirements.txt
-# or
-# create environment with conda
-conda env create -n ditto-env -f requirements.txt
+# create environment. Needed only the first time.
```
[the remark this answers is not in the document]
Comment (tkmamidi) [reply ReadMe.md:57]: I left it to the user to choose the environment name. They might already have `.venv` and want to create a new one?  Also, I think it is okay to have new files/directories being created while working with the repo a long as they're not submitting a PR for these things? 
I'm open to thoughts!

File: src/Home.py [hunk 5]
```diff
@@ -91,94 +114,93 @@ def main():
     explainer, clf = load_model()
 
     st.markdown("### Please input a variant of interest in build GRCh38:")
+    st.markdown("**Note:** Please use the correct variant info for the build. This app can hallucinate variants, but it is not intended for that purpose. You can check the correct variant info at [Ensembl](https://www.ensembl.org/index.html) or [UCSC](https://genome.ucsc.edu/index.html).")
```
[the remark this answers is not in the document]
Comment (tkmamidi) [reply src/Home.py:117]: There are downsides using openCravat's API. For example, it is tricky when trying out indels or giving out the same ref and alt alleles. Just some additional caution to the user won't hurt?

File: src/Home.py [hunk 6]
```diff
@@ -91,94 +114,93 @@ def main():
...
-        # SHAP explanation plot for a variant
-        pred_col2.pyplot(
-            shap.plots._waterfall.waterfall_legacy(
-                1 - explainer.expected_value[0],  # DITTO prediction for deleterious is (1 - y_pred)
-                np.negative(shap_value[0]),  # SHAP value for deleterious is negative
-                df2.values[0],
-                df2.columns,
-                max_display=20,
+    if ref != query_variant(str(chrom), int(pos)):
+        st.warning("Reference nucleotide does not match the reference genome. Please check the variant info.")
```
[the remark this answers is not in the document]
Comment (tkmamidi) [reply src/Home.py:128]: edited it accordingly.

File: src/Home.py [hunk 7]
```diff
@@ -91,94 +114,93 @@ def main():
...
+    else:
```
[the remark this answers is not in the document]
Comment (tkmamidi) [reply src/Home.py:129]: It felt good to see the banner, lol
Just a personal preference at this point!

File: src/Home.py [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+    st.markdown("\n")
+
+    st.markdown(
+        "### A tool for exploring genetic variants and their predicted functional impact."
+    )
+    st.markdown("- DITTO is a tool for exploring genetic variants and their predicted functional impact.")
+    st.markdown("- DITTO uses an explainable neural network model to predict the functional impact of variants and utilizes SHAP to explain the model's predictions.")
+    st.markdown("- DITTO provides annotations from OpenCravat, a tool for annotating variants with information from multiple data sources.")
+    st.markdown("- DITTO is currently trained on variants from ClinVar and is not intended for clinical use.")
+    st.markdown("- DITTO is a work in progress and we welcome your feedback and suggestions.")
```
[the remark this answers is not in the document]
Comment (tkmamidi) [reply src/Home.py:96]: I have a note just above the prediction to avoid redundancy.

File: ReadMe.md (hunk 3 above)
Comment (tkmamidi) [reply ReadMe.md:40]: Edited it but I'm not sure.

File: src/Home.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (ManavalanG) [reply src/Home.py:96]: Can you point me to it? Looks like I am missing something.

File: ReadMe.md (hunk 3 above)
Comment (ManavalanG) [reply ReadMe.md]: If you are not certain if it works in other OS, just say it was developed/tested in Mac. This way, we don't need to assume anything.

File: src/Home.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (tkmamidi) [reply src/Home.py:96]: It only shows up after submitting the variant for prediction.

File: ReadMe.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tkmamidi) [reply ReadMe.md]: Added an example just like we do with conda. Hope it helps!

**Comment by ManavalanG** [comment]
I came across a bug when I choose new transcript from `Choose a transcript:` dropdown. Instead of displaying results for that particular transcript, all of the results disappear instead.

**Comment by ManavalanG** [comment]
Transcript selection works as expected now, but when the webapp is opened, it shows error message - `AttributeError: st.session_state has no attribute "clicked". Did you forget to initialize it? More info: https://docs.streamlit.io/library/advanced-features/session-state#initialization`

**Review by ManavalanG — APPROVED** [review]

**Comment by ManavalanG** [comment]
Ready to be merged :)

**Pull request merged**